### PR TITLE
[chore] move carbon exporter to generated lifecycle tests

### PR DESCRIPTION
**Pull request by Frapschen**
Relates to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27849

### CODE REVIEW

**Comment by atoulme** [comment]
Please rebase

**Review by atoulme — APPROVED** [review]
LGTM

**Comment by atoulme** [comment]
Please rebase

**Review by atoulme — APPROVED** [review]
LGTM

**Comment by atoulme** [comment]
Pinging @aboguszewski-sumo for codeowner review

File: exporter/carbonexporter/metadata.yaml
Comment (aboguszewski-sumo) [inline exporter/carbonexporter/metadata.yaml:13]:
```suggestion
  expect_consumer_error: true
  
```

**Review by aboguszewski-sumo — APPROVED** [review]

File: exporter/carbonexporter/metadata.yaml
Comment (djaglowski) [inline exporter/carbonexporter/metadata.yaml:14]:
```suggestion

```

**Comment by atoulme** [comment]
We have the approvals, just need a rebase and updating collector versions.

**Comment by dmitryax** [comment]
@Frapschen can you please rebase?

**Review by dmitryax — APPROVED** [review]

**Pull request merged**
